### ISSUE TITLE
Language support?

### ISSUE THREAD

**Issue by OsnaFonz** [504rules/504rules.github.io#21]
Is it possible to make this little Gem language independent?

**Comment by Arachnid** [comment]
Yes! All the text is either in the template, or in the list of parts. If those are translated, it would be straightforward to choose between them - assuming the simpler option isn't just to copy and translate the whole thing.

What it needs is someone fluent in languages other than English to do the translation work.

**Comment by OsnaFonz** [comment]
Ok, i think that we wait until we think that the english version is nearly OK. Then i duplicate the index.html to index_de.html for example for a german version and begin to translate it.
alternatively we can put the template into an extra file. i tested this but then you can't open the file on an local computer.

**Comment by Arachnid** [comment]
Rather than putting the template in a separate file, we can use a separate 'script' tag, and choose which to use based on the language setting.

**Comment by bithalver** [comment]
Erm, maybe I do not understand all things, but if we duplicate the index.html then later we have to correct all mistakes in all copied index.hmtl files, right ?
Can we use instead some resource file with all the text ? That way the translation could be very easy (and does not need any understanding of code).
My two cents.

**Comment by Arachnid** [comment]
Indeed, that's why I'm suggesting embedding additional templates as preferable to duplicating the whole thing. Any errors in the templates will still need correcting in all languages, though.

**Comment by OsnaFonz** [comment]
if we use template files with a javascript load, we loose the possibility to open the generator on a local machine, because of security reasons. is this ok? 
and if this is ok, can we switch to the php-version of mustache? i think mustache is the main reason for the ios-problems (memory usage). and i think server-side rendering can fix this problem.

**Comment by Arachnid** [comment]
I'd rather keep it inline if we can. The template data isn't that big that we couldn't embed templates for multiple languages in one page; alternately we could fork the entire page, and put the javascript in its own file (which can be loaded on local FS).

I'm absolutely opposed to writing anything in PHP. :)

**Comment by someknave** [comment]
Let me know if it is worth creating some extra tokens, some languages have different plurals and genetive cases so using {{agent}}s, {{agent}}'s, {{owner}}'s, and {{owners}}' would not necessarily work in other languages, these contructions are fairly rare ({{agent}}s is the most common).
